### PR TITLE
Support setting CAS skip-nodes-with-custom-controller-pods parameter

### PR DESCRIPTION
**Pull request by shapirus**
Closes https://github.com/kubernetes/kops/issues/16057

### CODE REVIEW

File: upup/models/cloudup/resources/addons/cluster-autoscaler.addons.k8s.io/k8s-1.15.yaml.template [hunk 1]
```diff
@@ -344,6 +344,7 @@ spec:
             {{ end }}
             - --ignore-daemonsets-utilization={{ .IgnoreDaemonSetsUtilization }}
             - --scale-down-utilization-threshold={{ .ScaleDownUtilizationThreshold }}
+            - --skip-nodes-with-custom-controller-pods={{ .SkipNodesWithCustomControllerPods }}
```
Comment (hakman) [inline upup/models/cloudup/resources/addons/cluster-autoscaler.addons.k8s.io/k8s-1.15.yaml.template:347]: I think this was added in CAS 1.27 as part of https://github.com/kubernetes/autoscaler/commit/ff6fe5833d522d15670084bf996b3dea88231986.
```suggestion
            {{ if IsKubernetesGTE "1.27.0") }}
            - --skip-nodes-with-custom-controller-pods={{ .SkipNodesWithCustomControllerPods }}
            {{ end }}
```

**Review by hakman — CHANGES_REQUESTED** [review]

File: upup/models/cloudup/resources/addons/cluster-autoscaler.addons.k8s.io/k8s-1.15.yaml.template (hunk 1 above)
Comment (shapirus) [reply upup/models/cloudup/resources/addons/cluster-autoscaler.addons.k8s.io/k8s-1.15.yaml.template:348]: Right, it's a recent addition.

Comment (hakman) [reply upup/models/cloudup/resources/addons/cluster-autoscaler.addons.k8s.io/k8s-1.15.yaml.template:348]: I think there's a typo `)` and `hack/update-expected` should run again.

Comment (shapirus) [reply upup/models/cloudup/resources/addons/cluster-autoscaler.addons.k8s.io/k8s-1.15.yaml.template:348]: Yes true. Is it ok to overwrite the commit with a typo with a forced push or it'll be squashed on merge anyway? What's the accepted practice about this @hakman ?

Comment (hakman) [reply upup/models/cloudup/resources/addons/cluster-autoscaler.addons.k8s.io/k8s-1.15.yaml.template:348]: Please squash everything.

Comment (shapirus) [reply upup/models/cloudup/resources/addons/cluster-autoscaler.addons.k8s.io/k8s-1.15.yaml.template:348]: ...ok, squashed, force pushed, let's try again :)

**Comment by hakman** [comment]
Thanks, looks good. The failing test is broken but should be fixed by https://github.com/kubernetes/kops/pull/16061.

**Pull request merged**